docs/book/orders/cart-flow.rst

```diff
@@ -0,0 +1,47 @@
+.. index::
+   single: Cart flow
+
+Cart flow
+=========
+
+**Cart** in Sylius is an **Order** in the state ``cart``.
+It represents an order that is in progress (not placed yet).
+
+.. note::
+    In Sylius each user has their own cart. It can be cleared either by placing an order, removing items manually
+    or using cart clearing command.
+
+There are several cart flows, depending on the user being logged in or what items are currently placed in the cart.
+
+First scenario::
+
+    Given there is a not logged in user
+    And this user adds some items to the cart
+    And there is a customer identified by email "sylius@example.com" with not empty cart
+    When the not logged in user logs in using "sylius@example.com" email
+    Then the cart created by a not logged in user should be dropped
+    And the cart previously created by the user identified by "sylius@example.com" should be set as the current one
+
+Second scenario::
+
+    Given there is a not logged in user
+    And this user adds some items to the cart
+    And there is a customer identified by email "sylius@example.com" with an empty cart
+    When the not logged in user logs in using "sylius@example.com" email
+    Then the cart created by a not logged in user should not be dropped
+    And it should be set as the current cart
+
+Third scenario::
+
+    Given there is a customer identified by email "sylius@example.com" with an empty cart
+    And this user adds some items to the cart
+    When the user logs out
+    Then their cart should be dropped
+
+.. note::
+    The cart mentioned in the last scenario will we available once you log in again.
```

Comment:
Let's maybe prolong this scenario with logging in once again instead of this note?

Reply:
My point here was to make the scenario more coherent and therefore avoid multiple `When` and `Then` sections - wdyt?

Reply:
no problem with that :)